internal/asm/arm64/consts.go

```diff
@@ -34,305 +50,389 @@ const (
 const (
 	// Integer registers.
 
-	REG_R0 asm.Register = asm.NilRegister + 1 + iota
-	REG_R1
-	REG_R2
-	REG_R3
-	REG_R4
-	REG_R5
-	REG_R6
-	REG_R7
-	REG_R8
-	REG_R9
-	REG_R10
-	REG_R11
-	REG_R12
-	REG_R13
-	REG_R14
-	REG_R15
-	REG_R16
-	REG_R17
-	REG_R18
-	REG_R19
-	REG_R20
-	REG_R21
-	REG_R22
-	REG_R23
-	REG_R24
-	REG_R25
-	REG_R26
-	REG_R27
-	REG_R28
-	REG_R29
-	REG_R30
-	REGZERO
+	// RegR0 is the R0 register
+	RegR0 asm.Register = asm.NilRegister + 1 + iota
+	// RegR1 is the R1 register
+	RegR1
+	// RegR2 is the R2 register
+	RegR2
+	// RegR3 is the R3 register
+	RegR3
+	// RegR4 is the R4 register
+	RegR4
+	// RegR5 is the R5 register
+	RegR5
+	// RegR6 is the R6 register
+	RegR6
+	// RegR7 is the R7 register
+	RegR7
+	// RegR8 is the R8 register
+	RegR8
+	// RegR9 is the R9 register
+	RegR9
+	// RegR10 is the R10 register
+	RegR10
+	// RegR11 is the R11 register
+	RegR11
+	// RegR12 is the R12 register
+	RegR12
+	// RegR13 is the R13 register
+	RegR13
+	// RegR14 is the R14 register
+	RegR14
+	// RegR15 is the R15 register
+	RegR15
+	// RegR16 is the R16 register
+	RegR16
+	// RegR17 is the R17 register
+	RegR17
+	// RegR18 is the R18 register
+	RegR18
+	// RegR19 is the R19 register
+	RegR19
+	// RegR20 is the R20 register
+	RegR20
+	// RegR21 is the R21 register
+	RegR21
+	// RegR22 is the R22 register
+	RegR22
+	// RegR23 is the R23 register
+	RegR23
+	// RegR24 is the R24 register
+	RegR24
+	// RegR25 is the R25 register
+	RegR25
+	// RegR26 is the R26 register
+	RegR26
+	// RegR27 is the R27 register
+	RegR27
+	// RegR28 is the R28 register
+	RegR28
+	// RegR29 is the R29 register
+	RegR29
+	// RegR30 is the R30 register
+	RegR30
+	// RegZERO is the RZR register (read-only, always returning zero)
```

Comment:
Would it make sense to rename this to `RegRZR`?

Reply:
yeah makesense ! go for it